### ISSUE TITLE
v0.3.0 Stuck in replacement loop

### ISSUE THREAD

**Issue by archetyped** [bhughes339/vscode-replacerules#23]
Many existing rules are exhibiting strange behavior as of v0.3.0.  For example, the previously-working rule below now results in an infinite loop of the text replacement, which can only be stopped by force closing the document.

## Rule

```json
"Regex:Escape (JS)": {
    "find": "(\\\\|\")",
    "replace": "\\$1"
},
```

## Example Text

```
\sHello\b
```

### Expected

```
\\sHello\\b
```

### Actual (v0.3.0)

```
\\\\\\\\\\\\\\\\\\\\\\\\\\\\\\\\\\\\\\\\\\\\\\\\\\\\\\\\\\\\\\\\\\\\\\\\\\\\\\\\\\\\\\\\\\\\\\\\\\\\\\\\\\\\\\\\\\\\\\\\\\\\\\\\\\\\\\\\\\\\\\\\\\\\\\\\\\\\\\\\\\\\\\\\\\\\\\\\\\\\\\\\\\\\\\\\\\\\\\\\\\\\\\\\\\\\\\\\\\\\\\\\\\\\\\\\\\\\\\\\\\\\\\\\\\\\\\\\\\\\\\\\\\\\\\\\\\\\\\\\\\\\\\\\\\\\\\\\\\\\\\\
```

Note: Above text continues growing until document closed.

**Comment by bhughes339** [comment]
Thanks, I had a feeling there would be a bug with the new literal search/replace logic. Fix will be deployed momentarily.

**Comment by bhughes339** [comment]
Fixed in v0.3.1

**Comment by archetyped** [comment]
Thanks! 👍 Works as expected in v0.3.1.
